posts/2022-01-13-Rust-1.58.0.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: post
+title: "Announcing Rust 1.58.0"
+author: The Rust Release Team
+release: true
+---
+
+The Rust team is happy to announce a new version of Rust, 1.58.0.
+Rust is a programming language empowering everyone to build reliable and efficient software.
+
+If you have a previous version of Rust installed via rustup, getting Rust 1.58.0 is as easy as:
+
+```console
+rustup update stable
+```
+
+If you don't have it already, you can [get `rustup`][install]
+from the appropriate page on our website, and check out the
+[detailed release notes for 1.58.0][notes] on GitHub.
+
+[install]: https://www.rust-lang.org/install.html
+[notes]: https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1580-2022-01-13
+
+## What's in 1.58.0 stable
+
+Rust 1.58 brings captured identifiers in format strings, a change to the
+`Command` search path on Windows, more `#[must_use]` annotations in the
+standard library, and some new library stabilizations.
+
+### Captured identifiers in format strings
+
+Format strings can now capture arguments simply by writing `{ident}` in the
+string. Formats have long accepted positional arguments (optionally by index)
+and named arguments, for example:
+
+```rust
+println!("Hello, {}!", get_person());                // implicit position
+println!("Hello, {0}!", get_person());               // explicit index
+println!("Hello, {person}!", person = get_person()); // named
+```
+
+Now named arguments can also be captured from the surrounding scope, like:
+
+```rust
+let person = get_person();
+// ...
+println!("Hello, {person}!"); // captures the local `person`
+```
+
+This may also be used in formatting parameters:
+
+```rust
+let (width, precision) = get_format();
+for (name, score) in get_scores() {
+  println!("{name}: {score:width$.precision$}");
+}
+```
+
+Format strings can only capture plain identifiers, not arbitrary paths or
+expressions. For more complicated arguments, either assign them to a local name
+first, or use the older `name = expression` style of formatting arguments.
+
+This feature works in all macros accepting format strings. However, one corner
+case is the `panic!` macro in 2015 and 2018 editions, where `panic!("{ident}")`
+is still treated as an unformatted string -- the compiler will warn about this
+not having the intended effect.
```

Comment:
```suggestion
not having the intended effect. Due to the 2021 edition's update of panic
macros for [improved consistency], this works as expected in 2021 `panic!`.

[improved consistency]: https://doc.rust-lang.org/stable/edition-guide/rust-2021/panic-macro-consistency.html
```

Otherwise if you're not familiar with editions you may not know that at least new code doesn't need to worry about this today. (IOW, that an edition exists other than 2015/2018).